src/h3/_cy/unstable_vect.pyx

```diff
@@ -57,3 +57,17 @@ cpdef void geo_to_h3_vect(
         for i in range(len(lat)):
             c = deg2coord(lat[i], lng[i])
             out[i] = h3lib.geoToH3(&c, res)
+
+@boundscheck(False)
+@wraparound(False)
+cpdef void h3_to_parent_vect(
+    const H3Index[:] h,
+    int res,
+    H3Index[:] out
+):
```

Comment:
I'm not a pro at Cython, nor do I understand the H3 C API too well. When I try setting `nogil` like so:
```py
@boundscheck(False)
@wraparound(False)
cpdef void h3_to_parent_vect(
    const H3Index[:] h,
    int res,
    H3Index[:] out
) nogil:

    with nogil:
        for i in range(len(h)):
            out[i] = h3lib.h3ToParent(h[i], res)
```

I get a compilation error. Maybe I'm doing something wrong, but regardless vectorized with GIL is still a huge improvement over non-vectorized.
```
    Error compiling Cython file:
    ------------------------------------------------------------
    ...
        H3Index[:] out
    ) nogil:

        with nogil:
            for i in range(len(h)):
                out[i] = h3lib.h3ToParent(h[i], res)
                                        ^
    ------------------------------------------------------------

    /Users/kyle/github/mapping/h3-py/src/h3/_cy/unstable_vect.pyx:71:37: Calling gil-requiring function not allowed without gil
    gmake[2]: *** [src/h3/_cy/CMakeFiles/unstable_vect.dir/build.make:82: src/h3/_cy/unstable_vect.c] Error 1
    gmake[2]: *** Deleting file 'src/h3/_cy/unstable_vect.c'
    gmake[1]: *** [CMakeFiles/Makefile2:437: src/h3/_cy/CMakeFiles/unstable_vect.dir/all] Error 2
    gmake: *** [Makefile:171: all] Error 2
      File "/private/var/folders/6v/4_10tk8549bc7yrbycgd2c540000gn/T/pip-build-env-1vfe0biv/overlay/lib/python3.7/site-packages/skbuild/setuptools_wrap.py", line 589, in setup
        cmkr.make(make_args, env=env)
      File "/private/var/folders/6v/4_10tk8549bc7yrbycgd2c540000gn/T/pip-build-env-1vfe0biv/overlay/lib/python3.7/site-packages/skbuild/cmaker.py", line 507, in make
        os.path.abspath(CMAKE_BUILD_DIR())))
    An error occurred while building with CMake.
      Command:
        cmake --build . --target install --config Release --
      Source directory:
        /Users/kyle/github/mapping/h3-py
      Working directory:
        /Users/kyle/github/mapping/h3-py/_skbuild/macosx-10.15-x86_64-3.7/cmake-build
    Please see CMake's output for more information.
```

Reply:
This can be fixed by adding `nogil` to `h3ToParent` in `h3lib.pxd`.

Reply:
Yeah, that's the right fix!

Adding `nogil` to function declarations just tells Cython that those functions **can** be used without the GIL, but it won't automatically release the GIL when they're called.

That's what you're doing with

```python
cpdef void h3_to_parent_vect(
    const H3Index[:] h,
    int res,
    H3Index[:] out
) nogil:
```
and when you're adding `nogil` to `h3ToParent` in `h3lib.pxd`; just saying these functions **can** be used without having the GIL.

The `with nogil` block is different. It is actually telling Cython to release and re-acquire the GIL before and after that block. That's what your error is about above: you're releasing the GIL, but as far as Cython knows (before you added `nogil`), `h3ToParent` requires the GIL.

Reply:
Yeah that makes a ton of sense because presumably there will be functions that partially interact with Python and that can only release the GIL for portions of the function.